dali/pipeline/basic/crop.h

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_BASIC_CROP_H_
+#define DALI_PIPELINE_BASIC_CROP_H_
+
+#include <tuple>
+
+#include "dali/pipeline/basic/coords.h"
+#include "dali/pipeline/basic/runner.h"
+#include "dali/pipeline/basic/sequence.h"
+#include "dali/pipeline/basic/tensor.h"
+
+namespace dali {
+namespace basic {
+
+// TODO(klecki) allow for Schema to be present without using concrete types
+struct CropSchema {
+  using AllowedInputs = std::tuple<uint8_t>;
+  using AllowedOutputs = std::tuple<uint8_t, int8_t, int16_t, int32_t, int64_t, float>;
+};
+
+template <typename T, typename U, typename OutShape>
```

Comment:
In such cases, I would prefer more descriptive declaration of given types, (e.g. `InputType` or sth)